motion_computation/src/psm_to_external_object_convertor.cpp

```diff
@@ -401,47 +446,44 @@ rclcpp::Time get_psm_timestamp(const carma_v2x_msgs::msg::PSM& in_msg) {
   // then it can be assumed the initial_position is the same as the PSM data and
   // the utc_time can be used instead of sec_mark
   // clang-format off
-  if ((in_msg.presence_vector & 
-        carma_v2x_msgs::msg::PSM::HAS_PATH_HISTORY) &&
-      (in_msg.path_history.presence_vector & 
-        carma_v2x_msgs::msg::PathHistory::HAS_INITIAL_POSITION) &&
-      (in_msg.path_history.initial_position.presence_vector & 
-        carma_v2x_msgs::msg::FullPositionVector::HAS_UTC_TIME) &&
-      (in_msg.path_history.initial_position.utc_time.presence_vector &
-        carma_v2x_msgs::msg::FullPositionVector::YEAR &
-        carma_v2x_msgs::msg::FullPositionVector::MONTH &
-        carma_v2x_msgs::msg::FullPositionVector::DAY &
-        carma_v2x_msgs::msg::FullPositionVector::HOUR &
-        carma_v2x_msgs::msg::FullPositionVector::MINUTE &
-        carma_v2x_msgs::msg::FullPositionVector::SECOND) &&
-      in_msg.sec_mark.millisecond == in_msg.path_history.initial_position.utc_time.second) {
+  if ((in_msg.path_history.initial_position.utc_time.presence_vector & j2735_v2x_msgs::msg::DDateTime::YEAR) &&
+        (in_msg.path_history.initial_position.utc_time.presence_vector & j2735_v2x_msgs::msg::DDateTime::MONTH) &&
+        (in_msg.path_history.initial_position.utc_time.presence_vector & j2735_v2x_msgs::msg::DDateTime::DAY) &&
+        (in_msg.path_history.initial_position.utc_time.presence_vector & j2735_v2x_msgs::msg::DDateTime::HOUR) &&
+        (in_msg.path_history.initial_position.utc_time.presence_vector & j2735_v2x_msgs::msg::DDateTime::MINUTE) &&
+        (in_msg.path_history.initial_position.utc_time.presence_vector & j2735_v2x_msgs::msg::DDateTime::SECOND) &&
+      (in_msg.sec_mark.millisecond == in_msg.path_history.initial_position.utc_time.second.millisecond)) {
```

Comment:
Why change the structure of this check? Where is the validation that path_history and initial_position are even populated? Also the presence vector check seems logically the same but is more complex now since the bit operators are no longer chained.

Reply:
The check for path_history and initial_position was removed by mistake. But having the presence vector check in a chain was causing the if condition check to fail, even though all the presence vectors were set.

Reply:
Hmm that's pretty strange. I guess this is fine then